packages/acs-ui-common/src/safeStringify.ts

```diff
@@ -20,3 +20,20 @@ export const _safeJSONStringify = (
     return undefined;
   }
 };
+
+// Log all visited refs to avoid circular ref
+const createSafeReplacer = (): (<T>(key: string, value: T) => T | string) => {
+  const visited = new Set();
+  return function replacer<T>(key: string, value: T): T | string {
+    if (typeof value !== 'object') {
+      return value;
+    }
+
+    if (visited.has(value)) {
+      return 'Visited-Ref';
```

Comment:
Would be nice if we remembered the path to each object in `visited: Record<Object: string>` and then returned something like `Already visited at ${visited.get(value)}` so that you can still know what the value was.

Reply:
Good idea, will try to get it work tmr

Reply:
Just realize we don't have a good way to access current path... We might need to find a way to manipulate some keys+sub-keys+more to get it work, which might be tricky since we don't know which order stringify is visiting the node(Dfs or BFS)
I believe there are solutions, but a little overkill to this, since peple can guess the value from current key(value must be in ancestors)